test/client-unit/services/util.spec.js

```diff
@@ -74,4 +74,130 @@ describe('util', () => {
+    const form = {};
+    const expected = {};
+    const elements = util.filterFormElements(form);
+    expect(elements).to.deep.equal(expected);
+  });
+
+  it.skip('#filterFormElements() should return $modelValue or $bhValue properties of sub-objects', () => {
+
+  });
+
+  it('#clean() should filter out all properties that have empty values or that begin with $', () => {
+    const data = {
+      $$hashKey : '8339Gh1',
+      $modelValue : 10,
+      $dirty : true,
+      name : 'Alice',
+      test : null,
+    };
+
+    const expected = {
+      $modelValue : 10,
+      $dirty : true,
+      name : 'Alice',
+    };
+    const formatedData = util.clean(data);
+    expect(formatedData).to.deep.equal(expected);
+  });
+
+  it('#getMomentAge() should return a number', () => {
+    const date = new Date();
+    const expected = 'number';
+    const result = util.getMomentAge(date);
+    expect(typeof result).to.be.equal(expected);
+  });
+
+  it('#defaultBirthMonth should be defined', () => {
+    const { defaultBirthMonth } = util;
+    expect(defaultBirthMonth).to.not.equal(null);
+  });
+
+  it('#uniquelize should turn an array into unique values', () => {
+    const data = ['name', 'gender', 'label', 'name', 'gender'];
+    const expected = ['name', 'gender', 'label'];
+    const formatedData = util.uniquelize(data);
+    expect(formatedData).to.deep.equal(expected);
+  });
+
+  it('#isEmptyObject() should turn true for {}', () => {
+    const data = {};
+    const expected = true;
+    const formatedData = util.isEmptyObject(data);
+    expect(formatedData).to.be.equal(expected);
+  });
+
+  it('#isEmptyObject() should turn true for []', () => {
+    const data = [];
+    const expected = true;
+    const formatedData = util.isEmptyObject(data);
+    expect(formatedData).to.be.equal(expected);
+  });
+
+  it('#xor should returns the logical XOR of two booleans.', () => {
+    const expected = true;
+    const formatedData = util.xor(true, false);
+    expect(formatedData).to.be.equal(expected);
+  });
+
+  it('#maskObjectFromKeys should return an object with only the properties specified as a second value', () => {
+    const data = {
+      $$hashKey : '8339Gh1',
+      $modelValue : 10,
+      $dirty : true,
+      name : 'Alice',
+      age : 12,
+    };
+    const mask = ['name', 'age'];
+    const expected = { name : 'Alice', age : 12 };
+    const formatedData = util.maskObjectFromKeys(data, mask);
+    expect(formatedData).to.deep.equal(expected);
+  });
+
+  it('#maskObjectFromKeys should return an empty object if given empty keys', () => {
+    const data = {
+      $modelValue : 10,
+      $dirty : true,
+      name : 'Alice',
+      age : 12,
+    };
+    const mask = [];
+    const expected = {};
+    const formatedData = util.maskObjectFromKeys(data, mask);
+    expect(formatedData).to.deep.equal(expected);
+  });
+
+
+  it.skip('debounce() should fire a function in the future.', () => {
```

Comment:
The way to test this is with a [spy](https://www.chaijs.com/plugins/chai-spies/).  We need to add a function to debounce, and check that the function is called later.

Since [debounce() uses a timeout](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/util.js#L270), we will need to do the same thing here:
```js
it('#debounce() should fire a function in the future', (done) => {
  const spy = chai.spy();
  const debounced = debounce(spy, 200);  // debounce spy for 200 milliseconds
  expect(spy).to.not.have.been.called();
  setTimeout(() => {
    expect(spy).to.have.been.called();
    done();
   }, 250);
});
```

The basic idea is:
  1. We create a spy function that tells us when it is called or not.
  2. We pass in the `done()` function so that [mocha knows that this test is asynchronous](https://stackoverflow.com/a/21175342).
   3. We make sure the function is not called immediately (otherwise it might a problem with the function).
   4. We make sure the function is called after the timeout.